src/Apps/Legion/Components/WizardShell/Internal/DataSourceStep/DataSourceStep.types.ts

```diff
@@ -15,6 +15,8 @@ import { IReactSelectOption } from '../../../DataPusher/DataPusher.types';
 
 export interface IDataSourceStepProps {
     adapter: IDataManagementAdapter;
+    registerNextButtonClick: (fn: VoidFunction) => void;
```

Comment:
thoughts on pushing this to the context? then we can handle the seconary actions as well.